### PR TITLE
Prepare release 0.10.0beta4

### PR DESCRIPTION
**Pull request by notoraptor**
@nouiz 

NB: Vision / State not updated (should we mention here that theano development will be stopped ?).

### CODE REVIEW

File: HISTORY.txt [hunk 1]
```diff
@@ -5,9 +5,114 @@
...
+Theano 0.10.0beta3 (20th of September, 2017)
+============================================
+
+This release contains new features, improvements and bug fixes to prepare the upcoming release candidate.
+
+We recommend that every developer updates to this version.
+
+Highlights:
+ - conda packages now available and updated in our own conda channel ``mila-udem``.
+   To install it: ``conda install -c mila-udem/label/pre theano pygpu``
```
Comment (nouiz) [inline HISTORY.txt]: This should be:

conda install -c mila-udem/label/pre -c mila-udem theano pygpu

to have the latest release of pygpu

File: HISTORY.txt [hunk 2]
```diff
@@ -5,9 +5,114 @@
...
+
+ - Improved elemwise operations
+
+   - Speed-up elemwise ops based on SciPy
+   - Fixed memory leak related to elemwise ops on GPU
+
+ - Improved pickling and tests in debug mode
```
Comment (nouiz) [inline HISTORY.txt]: Remove this line. Not useful

File: NEWS.txt [hunk 3]
```diff
@@ -3,42 +3,40 @@ Release Notes
...
+ - Implemented truncated normal distribution with box-muller transform
+ - Added ``L_op()`` overriding option for ``OpFromGraph``
+ - Added NumPy C-API based fallback implementation for ``[sd]gemv_`` and ``[sd]dot_``
 
-Others:
- - Added ``R_op()`` for ``ZeroGrad``
- - Added description for rnnblock
+Other more detailed changes:
+ - Improved stack trace follow-up for GPU optimizations
+ - Fixed gradient error for elemwise ``minimum`` and ``maximum``
```
Comment (nouiz) [inline NEWS.txt]: specify when both values compared are the same.

File: NEWS_DEV.txt [hunk 4]
```diff
@@ -115,6 +122,9 @@ GPU:
    - Started to use ``float32`` precision for computations that don't support ``float16`` on GPU
 
 New features:
+ - Implemented truncated normal distribution with box-muller transform
+ - Added ``L_op()`` overriding option for ``OpFromGraph``
```
Comment (nouiz) [inline NEWS_DEV.txt]: where this come from?

Comment (notoraptor) [reply NEWS_DEV.txt]: From here: https://github.com/Theano/Theano/pull/5641/

Comment (nouiz) [reply NEWS_DEV.txt]: ok

File: NEWS.txt (hunk 3 above)
Comment (notoraptor) [reply NEWS.txt]: Fixed

File: HISTORY.txt (hunk 2 above)
Comment (notoraptor) [reply HISTORY.txt]: Fixed

File: HISTORY.txt (hunk 1 above)
Comment (notoraptor) [reply HISTORY.txt]: Fixed

**Pull request merged**